test/com/google/enterprise/adaptor/filenet/FileNetAdaptorTest.java

```diff
@@ -73,4 +97,282 @@ public void testGetConnection() throws Exception {
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("foo/bar")),
+        response);
+    assertEquals(RecordingResponse.State.NOT_FOUND, response.getState());
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testGetDocContent_invalidDocId3() throws Exception {
+    adaptor.init(context);
+
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(FileNetAdaptor.class, "Unsupported type:", messages);
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("pseudo/type=foo")),
+        response);
+    assertEquals(RecordingResponse.State.NOT_FOUND, response.getState());
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testGetDocContent_shortCheckpoint() throws Exception {
+    config.overrideKey("feed.maxUrls", "5");
+    adaptor.init(context);
+
+    long startTime = new Date().getTime();
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("pseudo/type=document")),
+        response);
+
+    List<Record> golden = ImmutableList.of(
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000001}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000002}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000003}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000004}"))
+        .build());
+
+    RecordingDocIdPusher pusher
+        = (RecordingDocIdPusher) context.getDocIdPusher();
+    List<Record> actual = pusher.getRecords();
+    // Assert that the pushed DocIds match.
+    assertEquals(golden, actual.subList(0, actual.size() - 1));
+
+    // Now check the continuation record.
+    Record continuation = actual.get(actual.size() - 1);
+    String docid = continuation.getDocId().getUniqueId();
+    assertTrue(docid, docid.startsWith("pseudo/"));
+    Checkpoint checkpoint
+        = new Checkpoint(docid.substring(docid.indexOf('/') + 1));
+    assertEquals("document", checkpoint.type);
+    assertEquals(new Id("{AAAAAAAA-0000-0000-0000-000000000004}"),
+                 checkpoint.guid);
+    assertTrue((checkpoint.timestamp.getTime() - startTime) < 2000);
+  }
+
+  @Test
+  public void testGetDocContent_longCheckpoint() throws Exception {
+    config.overrideKey("feed.maxUrls", "5");
+    adaptor.init(context);
+
+    Date timestamp = new Date();
+    Checkpoint startCheckpoint = new Checkpoint("document", timestamp,
+        new Id("{AAAAAAAA-0000-0000-0000-000000000004}"));
+
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(
+        new MockRequest(new DocId("pseudo/" + startCheckpoint)), response);
```

Comment:
Maybe newDocId? I was also tempted by a subclass, maybe even called "DocId" so that you can't do the wrong thing. Not sold on that idea, just seemed worth pondering.